pandas/tests/io/json/test_pandas.py

```diff
@@ -985,12 +987,36 @@ def test_tz_range_is_utc(self):
         df = DataFrame({'DT': dti})
         assert dumps(df, iso_dates=True) == dfexp
 
-    def test_read_jsonl(self):
+    def test_read_inline_jsonl(self):
         # GH9180
         result = read_json('{"a": 1, "b": 2}\n{"b":2, "a" :1}\n', lines=True)
         expected = DataFrame([[1, 2], [1, 2]], columns=['a', 'b'])
         assert_frame_equal(result, expected)
 
+    @mock_s3
+    def test_read_s3_jsonl(self):
+        # GH17200
+        conn = boto3.client('s3')
```

Comment:
This setup stuff can be moved to a pytest fixture. Something like


```python
@mock_s3
@pytest.fixture
def testbucket:
    """Fixture for ..."""
    conn = ...
```

Then your `test_read_s3_jsonl` will take a `testbucket` parameter. I don't know if it will have to be decorated by `mock_s3`, but I *think* it will.

Reply:
Also move the boto3 import here and use `boto3 = pytest.importorskip('boto3')` so that tests trying to use the fixture will skip if boto3 isn't installed.